logger_test.go

```diff
@@ -371,12 +327,44 @@ func TestJSONLoggerSyncsOutput(t *testing.T) {
 }
 
 func TestLoggerAddCaller(t *testing.T) {
-	withJSONLogger(t, DebugLevel, opts(AddCaller()), func(logger Logger, buf *testBuffer) {
-		logger.Info("Callers.")
+	sink := &testBuffer{}
+	errSink := &testBuffer{}
+
+	log1 := New(
+		WriterFacility(newJSONEncoder(NoTime()), sink, DebugLevel),
+		ErrorOutput(errSink), AddCaller())
+	log2 := New(
+		WriterFacility(newJSONEncoder(NoTime()), sink, DebugLevel),
+		ErrorOutput(errSink), AddCaller(), AddCallerSkip(1))
+	log3 := New(
+		WriterFacility(newJSONEncoder(NoTime()), sink, DebugLevel),
+		ErrorOutput(errSink), AddCaller(), AddCallerSkip(1), AddCallerSkip(1))
+	// TODO: wrap(logger) family to be tested by upcoming sugared logger
+
+	log1.Info("mess1")
+	log2.Info("mess2")
+	log3.Info("mess3")
+	log1.Check(InfoLevel, "mess4").Write()
+	log2.Check(InfoLevel, "mess5").Write()
+	log3.Check(InfoLevel, "mess6").Write()
+
+	lines := sink.Lines()
+	for i, testCase := range []struct{ pat, name string }{
+		{`"caller":"[^"]+/logger_test.go:\d+","msg":"mess1"`, "log1.Info"},
+		{`"caller":"[^"]+/testing.go:\d+","msg":"mess2"`, "log2.Info"},
+		{`"caller":"[^"]+/src/runtime/.*:\d+","msg":"mess3"`, "log3.Info"},
+		{`"caller":"[^"]+/logger_test.go:\d+","msg":"mess4"`, "log1.Check(Info)"},
+		{`"caller":"[^"]+/testing.go:\d+","msg":"mess5"`, "log2.Check(Info)"},
+		{`"caller":"[^"]+/src/runtime/.*:\d+","msg":"mess6"`, "log3.Check(Info)"},
+	} {
 		assert.Regexp(t,
-			`"caller":"[^"]+/logger_test.go:[\d]+","msg":"Callers\."`,
-			buf.Stripped(), "Expected to find package name and file name in output.")
-	})
+			testCase.pat,
+			lines[i],
+			"Expected to find package name and file name in %s output.",
+			testCase.name)
+	}
+
+	assert.Empty(t, errSink.String(), "Expected error sink to be empty.")
```

Comment:
Can we structure this so that each test case includes the logger and the expected message, and in the test loop we

```
tt.logger.Info("foo")
tt.logger.Check(InfoLevel, "foo").Write()
```

and then assert that we got the expected message twice? Seems like that would more clearly associate the loggers and the expectations.